summingbird-storm/src/test/scala/com/twitter/summingbird/storm/TestKeyValueSpout.scala

```diff
@@ -0,0 +1,152 @@
+  }
+
+  "Check two batches are being sent - 4 different tuples" in {
+    val summer = TestKeyValueSpout.getSyncSummingQueueBuildSummer(1, 1, 1)
+    val timeStamp = Timestamp.now
+
+    //input
+    val basespout = new BaseSpout[(Timestamp, (Int, Int))] {
+      override def poll = scala.collection.immutable.List((timeStamp, (1, 1)), (timeStamp, (2, 1)), (timeStamp, (3, 1)), (timeStamp, (4, 1))).toSeq
+    }
+
+    //output
+    val expectedTuples = MSet[(Int, Map[_, _])]()
+    expectedTuples.add((0, Map((1, BatchID(1)) -> (timeStamp, 1), (2, BatchID(1)) -> (timeStamp, 1))))
+    expectedTuples.add((0, Map((3, BatchID(1)) -> (timeStamp, 1), (4, BatchID(1)) -> (timeStamp, 1))))
+
+    val (spout, collector) = process(basespout, summer, expectedTuples)
+    spout.nextTuple()
+    assert(collector.checkSize == 0)
+  }
+
+  "Tuples are to be aggregated - all four tuples should be crushed down" in {
+    val summer = TestKeyValueSpout.getSyncSummingQueueBuildSummer(2, 1, 1)
+    val timeStamp = Timestamp.now
+
+    //input
+    val basespout = new BaseSpout[(Timestamp, (Int, Int))] {
+      override def poll = scala.collection.immutable.List((timeStamp, (1, 1)), (timeStamp, (1, 2)), (timeStamp, (1, 3)), (timeStamp, (1, 4))).toSeq
+    }
+
+    //output
+    val expectedTuples = MSet[(Int, Map[_, _])]()
+    expectedTuples.add((0, Map((1, BatchID(1)) -> (timeStamp, 6))))
+
+    val (spout, collector) = process(basespout, summer, expectedTuples)
+    spout.nextTuple()
+    assert(collector.checkSize == 0)
+  }
+
+  "No tuples should be emitted without timerflush - batch size greater than number of tuples" in {
+    val summer = TestKeyValueSpout.getSyncSummingQueueBuildSummer(10, 1, 1)
+    val timeStamp = Timestamp.now
+
+    //input
+    val basespout = new BaseSpout[(Timestamp, (Int, Int))] {
+      override def poll = scala.collection.immutable.List((timeStamp, (1, 1)), (timeStamp, (2, 1)), (timeStamp, (3, 1)), (timeStamp, (4, 1))).toSeq
+    }
+
+    //output
+    val expectedTuples = MSet[(Int, Map[_, _])]()
+
+    val (spout, collector) = process(basespout, summer, expectedTuples)
+    spout.nextTuple()
+    assert(collector.checkSize == 0)
+  }
+
+  "One batch should be sent on timerFlush - batch size greater than number of tuples" in {
+    val summer = TestKeyValueSpout.getSyncSummingQueueBuildSummer(10, 1, 1)
+    val timeStamp = Timestamp.now
+
+    //input
+    val basespout = new BaseSpout[(Timestamp, (Int, Int))] {
+      override def poll = scala.collection.immutable.List((timeStamp, (1, 1)), (timeStamp, (2, 1)), (timeStamp, (3, 1)), (timeStamp, (4, 1)))
+    }
+
+    //output
+    val expectedTuples = MSet[(Int, Map[_, _])]()
+    expectedTuples.add((0, Map((2, BatchID(1)) -> (timeStamp, 1), (1, BatchID(1)) -> (timeStamp, 1), (3, BatchID(1)) -> (timeStamp, 1), (4, BatchID(1)) -> (timeStamp, 1))))
+
+    val (spout, collector) = process(basespout, summer, expectedTuples)
+    spout.nextTuple()
+    Thread.sleep(1000)
```

Comment:
I have made the flushFrequency as 1000 millis and waiting for 1200 millis. This should work.